api/core/v1beta2/clusterclass_types.go

```diff
@@ -160,14 +160,13 @@ type ControlPlaneClass struct {
 	// +required
 	TemplateRef ClusterClassTemplateReference `json:"templateRef,omitempty,omitzero"`
 
-	// machineInfrastructure defines the metadata and infrastructure information
-	// for control plane machines.
+	// infrastructureMachine defines infrastructure for control plane machines.
 	//
 	// This field is supported if and only if the control plane provider template
 	// referenced above is Machine based and supports setting replicas.
 	//
 	// +optional
-	MachineInfrastructure *ControlPlaneClassMachineInfrastructureTemplate `json:"machineInfrastructure,omitempty"`
+	InfrastructureMachine *ControlPlaneClassInfrastructureMachineTemplate `json:"infrastructureMachine,omitempty"`
```

Comment:
I think the idea was to also change infrastructure to infrastructureMachine in MachineDeploymentClass. Is this still the plan? (or is this just a separate PR?)